woocommerce/woocommerce
# Product Collection: Fix "Products to show" inspector control not showing as expected on Cart template

### Changes proposed in this Pull Request:





Closes #52171

This PR addresses an issue where the "Products to show" accordion item in the inspector was empty when adding an "Upsells" Collection to the Cart template. The fix ensures that the correct UI and options are displayed in the inspector, allowing users to choose between showing upsells from products in the cart or from a specific product.




### How to test the changes in this Pull Request:



**Case 1: Products to show accordion is visible with correct UI**

1. Go to `Page: Cart` template.
2. Add a Product Collection block and choose "Upsells" collection.
3. Verify that "PRODUCTS TO SHOW" accordion is visible with the correct UI as shown in screenshot below
![image](https://github.com/user-attachments/assets/410b8c2c-f24e-445b-bba3-3231ef2e8c01)

4. Verify that by default, "From products in the cart" option is selected.

**Case 2: Testing on Cart page**

1. Go to `Page: Cart` template.
2. Add a Product Collection block and choose "Upsells" collection.
3. Publish the page and go to frontend
4. Add a product with upsells to cart
5. Go to cart page
6. Verify that:
	- Collection is rendered with "You may also like" heading
	- Collection shows correct upsells products from products which are in cart
	- For more info, see PR #51374
7. Go back to Editor and change "From products in the cart" to "From a specific product" 
8. Select a specific product and save the page
9. Go to frontend and verify that on Cart page, upsells products are always shown from a specific product you selected in the inspector control. These products should not change when you add/remove other products to cart.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Product Collection: Fix "Products to show" inspector control not showing as expected on Cart template
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
